lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/launch/JavaLaunchCommandBuilder.java

```diff
@@ -25,17 +25,30 @@
 import java.io.File;
 import java.util.ArrayList;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public final class JavaLaunchCommandBuilder implements LaunchCommandBuilder {
+  private static final Logger LOG = Logger.getLogger(JavaLaunchCommandBuilder.class.getName());
+
   private static final String DEFAULT_JAVA_PATH = System.getenv("JAVA_HOME") + "/bin/" + "java";
   private String stderrPath = null;
   private String stdoutPath = null;
-  private int megaBytes = 0;
   private String evaluatorConfigurationPath = null;
   private String javaPath = null;
   private String classPath = null;
   private Boolean assertionsEnabled = null;
+  private Map<String, JVMOption> options = new HashMap<>();
+  private Map<String, JVMOption> defaultOptions = new HashMap<String, JVMOption>() {{
```

Comment:
> I'm uncomfortable with double-braces initialization

We don't have a policy on it. In this case, either way would be fine. However, there are cases where it is useful: Initializing an immutable collection comes to mind.

> Guava has nice syntax

Yes, but we worked hard to rid ourselves from the guava dependencies. Guava is breaking compatibility far too often for other libraries to depend on it. By not depending on it, we allow REEF applications to use the Guava version of their choosing. And yes, shading would be an alternative, but that creates challenges of its own.